add repo code metrics(nloc)

ckxkexing opened the pull request:
python requirement:
- [ ] lizard

ckxkexing committed:
add repo code metrics(nloc)

crystaldust commented on dags/oss_know/libs/metrics/repo_code_metrics.py:
```diff
@@ -0,0 +1,86 @@
+    def calculate_metrics(self):
+        logger.info(f'calculating by {self.owner}, {self.repo}')
+        dir = f"./{self.owner}/{self.repo}"
+        clone_one_depth(self.owner, self.repo, dir, True)
```
这里的机制可以设计成：
- 如果repo dir不存在，clone到本地
- 如果repo dir已存在，直接pull
- 不必每次计算完就删掉，可以利用本地存储空间换取网络传输的时间

crystaldust commented on dags/oss_know/libs/metrics/repo_code_metrics.py:
```diff
@@ -0,0 +1,86 @@
+
+    def routine_calculate_metrics_once(self):
+        metrics = self.calculate_metrics()
+        metrics_keys = ["owner", "repo", "created_at", "sha", "nloc", "average_nloc", "average_cyclomatic_complexity", "average_token_count"]
```
这些固定的字段，可以设计成类的static成员，不必每次都声明变量

ckxkexing committed:
use static variable

ckxkexing committed:
use pull if repo exist!

crystaldust committed:
Fix the refs to metrics routine related vars

crystaldust committed:
Update the git repos' path for repo code metric calculation.

This is a temporary tradeoff, the routine calculation should provide
a way to pass some extra parameters, so that specific routines can be
satisfied

crystaldust committed:
Optimize and correct repo code metrics calculation utils.

- Optimize imports and logs and code styles
- Handle the situation when the repo is unavailable
- Do not remove the repo after calculation

ckxkexing committed:
Merge pull request #1 from crystaldust/chenkexing-repo-code-metrics

Correct and improve repo code metrics calculation
